src/lib/index.ts

```diff
@@ -25,13 +23,14 @@ export interface IRequest {
 }
 
 const queue = () => {
-    const store = createStore({reducer, middlewares});
+    const store = createStore({ reducer, middlewares });
 
     const schedule = (manifest: IRequest) => {
         store.dispatch(
-            add(assign({}, manifest, {
+            add({
                 id: uuid(),
-            }))
+                ...manifest,
```

Comment:
If manifest owns an ID, ```id``` would be overwritten - intended?
If not, you have to change the order ```{ ...manifest, id: uuid() }```